### PR TITLE
Add amp training example document

### PR DESCRIPTION
**Pull request by huangxu96**
### PR types

New features
### PR changes

Docs
### Describe

Add docs of amp training. add an alias of AutoMixedPrecisionLists.

### CODE REVIEW

File: python/paddle/fluid/contrib/mixed_precision/decorator.py [hunk 1]
```diff
@@ -202,6 +202,54 @@ def amp_init(self,
...
+        Examples:
+            .. code-block:: python
+
+                import numpy as np
+                import paddle
+                import paddle.nn.functional as F
+
+                paddle.enable_static()
+
+                place = paddle.CUDAPlace(0) # paddle.CPUPlace()
```
Comment (wzzju) [inline python/paddle/fluid/contrib/mixed_precision/decorator.py]: Remove `CPUPlace`.

File: python/paddle/fluid/contrib/mixed_precision/decorator.py [hunk 2]
```diff
@@ -403,17 +451,82 @@ def decorate(optimizer,
...
+        .. code-block:: python
+
+            # pure fp16 training example
+            import numpy as np
+            import paddle
+            import paddle.nn.functional as F
+
+            paddle.enable_static()
+
+            place = paddle.CUDAPlace(0) # paddle.CPUPlace()
```
Comment (wzzju) [inline python/paddle/fluid/contrib/mixed_precision/decorator.py]: Remove `CPUPlace`.

File: python/paddle/fluid/contrib/mixed_precision/fp16_utils.py [hunk 3]
```diff
@@ -282,6 +282,28 @@ def fp16_guard():
...
+        .. code-block:: python
+
+            # pure fp16 training example
+            import numpy as np
+            import paddle
+            import paddle.nn.functional as F
+
+            paddle.enable_static()
+
+            place = paddle.CUDAPlace(0) # paddle.CPUPlace()
```
Comment (wzzju) [inline python/paddle/fluid/contrib/mixed_precision/fp16_utils.py]: Remove `CPUPlace`.

**Review by wzzju — CHANGES_REQUESTED** [review]

**Review by TCChenlong — APPROVED** [review]
LGTM

File: python/paddle/fluid/contrib/mixed_precision/decorator.py (hunk 2 above)
Comment (huangxu96) [reply python/paddle/fluid/contrib/mixed_precision/decorator.py]: Done.

File: python/paddle/fluid/contrib/mixed_precision/fp16_utils.py (hunk 3 above)
Comment (huangxu96) [reply python/paddle/fluid/contrib/mixed_precision/fp16_utils.py]: Done.

File: python/paddle/fluid/contrib/mixed_precision/decorator.py (hunk 1 above)
Comment (huangxu96) [reply python/paddle/fluid/contrib/mixed_precision/decorator.py]: Done.

**Review by wzzju — APPROVED** [review]
LGTM.

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**